drivers/sensor/bq274xx/bq274xx.c

```diff
@@ -367,14 +554,18 @@ static int gauge_init(const struct device *dev)
 	}
 #endif
 
-	ret = get_device_type(dev, &id);
+	ret = bq274xx_get_device_type(dev, &id);
 	if (ret < 0) {
 		LOG_ERR("Unable to get device ID");
 		return -EIO;
 	}
 
-	if (id != BQ274XX_DEVICE_ID) {
-		LOG_ERR("Invalid Device");
+	if (id == BQ27421_DEVICE_ID) {
+		data->regs = &bq27421_regs;
+	} else if (id == BQ27427_DEVICE_ID) {
+		data->regs = &bq27427_regs;
```

Comment:
Would it be better to do this at compile time to reduce flash usage?

Reply:
There seems to be quite a few variations to potentially support: https://lxr.missinglinkelectronics.com/linux/drivers/power/supply/bq27xxx_battery.c#L136

Reply:
Yeah I'm unsure about it, with these two we can get away with just 4 bytes of flash. If we introduce build time variations we'd have to go for part number specific compatibles and rework the driver a bit more and have downstream user change it in their application. I was thinking about trying to go with this, keep it simple for now and see if there's demand for the other chips in the family.

Reply:
It could easily be done based on compatible at instantiation time, but can be done later, this PR is already doing a good cleanup

Reply:
Actually I was starting to like the idea of keeping this in runtime: the two p/n right now are a pin-to-pin compatible, as it is now you can drop the new one and it should work with no code changes, if the cost for that is only few bytes of flash I'd say it may be worth it.